README.md

```diff
@@ -2,302 +2,708 @@
+end
 ```
 
-### :nullable fields
+## Merging schemas
+
+The `#merge` method will merge field definitions in two schemas and produce a new schema instance.
+
+```ruby
+basic_user_schema = Parametric::Schema.new do
+  field(:name).type(:string).required
+  field(:age).type(:integer)
+end
+
+friends_schema = Parametric::Schema.new do
+  field(:friends).type(:array).schema do
+    field(:name).required
+    field(:email).validate(:email)
+  end
+end
+
+user_with_friends_schema = basic_user_schema.merge(friends_schema)
 
-In some cases you won't want Parametric to provide nil or empty keys for attributes missing from the input. For example when missing keys have special meaning in your application.
+results = user_with_friends_schema.resolve(input)
+```
 
-In those cases you can add the `:nullable` option to said param definitions:
+Fields defined in the merged schema will override fields with the same name in the original schema.
 
 ```ruby
-class OrdersSearch
-  include Parametric::Params
-  param :query, 'Search query. optional', nullable: true
-  param :tags, 'Tags', multiple: true
+required_name_schema = Parametric::Schema.new do
+  field(:name).required
+  field(:age)
 end
 
-search = OrdersSearch.new({})
-search.params # {tags: []}
+optional_name_schema = Parametric::Schema.new do
+  field(:name)
+end
+
+# This schema now has :name and :age fields.
+# :name has been redefined to not be required.
+new_schema = required_name_schema.merge(optional_name_schema)
 ```
 
-## `available_params`
+## #meta
 
-`#available_params` returns the subset of keys that were populated (including defaults). Useful for building query strings.
+The `#meta` field method can be used to add custom meta data to field definitions.
+These meta data can be used later when instrospecting schemas (ie. to generate documentation or error notices).
 
 ```ruby
-order_search = OrdersSearch.new(page: 2, foo: 'bar')
-order_search.available_params # => {page: 2, per_page: 50}
+create_user_schema = Parametric::Schema.do
+  field(:name).required.type(:string).meta(label: "User's full name")
+  field(:status).options(["published", "unpublished"]).default("published")
+  field(:age).type(:integer).meta(label: "User's age")
+  field(:friends).type(:array).meta(label: "User friends").schema do
+    field(:name).type(:string).present.meta(label: "Friend full name")
+    field(:email).validate(:email).meta(label: "Friend's email")
+  end
+end
 ```
 
-## `schema`
+## #schema
 
-`#schema` returns a data structure including meta-data on each parameter, such as "label" and "options". Useful for building forms or self-documented Hypermedia APIs (or maybe [json-schema](http://json-schema.org/example2.html) endpoints).
+A `Schema` instance has a `#schema` method that allows instrospecting schema meta data.
 
 ```ruby
-order_search.schema[:q].label # => 'Full text search query'
-order_search.schema[:q].value # => ''
+create_user_schema.schema[:name].label # => "User's full name"
```

Comment:
The `.schema` method makes me feel a little bit uncomfortable.

1. Here it's used to get metadata out of an existing schema. When defining schemas, `.schema` declares a field to be a nested schema.
1. The syntax looks odd to me: `.schema[:key].method`, looks like I'm calling a method which returns a hash, I'm then getting a value out of the hash and then calling a method on it. I think it's unusual to go from object to hash to object. Usually when I've had to write code that looks like this before, it's been because I'm reaching into an undocumented part of a library, which was never designed for external use...